aztokenprovider/token_cache.go

```diff
@@ -67,10 +69,23 @@ func (c *tokenCacheImpl) getEntryFor(ctx context.Context, credential TokenRetrie
 	tid := returnGrafanaMultiTenantId(ctx)
 
 	key := credential.GetCacheKey(tid)
-	if entry, ok = c.cache.Load(key); !ok {
+
+	entry, ok = c.cache.Load(key)
+	// Store new cache value if there isn't an existing one
+	if !ok {
 		entry, _ = c.cache.LoadOrStore(key, &credentialCacheEntry{
 			retriever: credential,
 		})
+		return entry.(*credentialCacheEntry)
+	}
+
+	expiry := entry.(*credentialCacheEntry).retriever.GetExpiry()
```

Comment:
Could we move this logic into to `Load` of retriever obo so we don't implement a bunch of functions that return nil? So `GetExpiry` would just be a function and not a member function.

Reply:
Talked in slack. keeping the same if/when we implement expiration for other auth types